pkg/csi/service/wcp/controller.go

```diff
@@ -1636,23 +1636,61 @@ func (c *controller) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshot
 				return nil, logger.LogNewErrorCodef(log, codes.Internal,
 					"failed to retrieve cnsVolumeInfo for volume: %s Error: %+v", volumeID, err)
 			}
-			snapshotID, snapshotCreateTimePtr, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
+			snapshotID, cnsSnapshotInfo, err = common.CreateSnapshotUtil(ctx, c.manager.VolumeManager,
 				volumeID, req.Name, &cnsvolume.CreateSnapshotExtraParams{
 					StorageClassName:           cnsVolumeInfo.Spec.StorageClassName,
 					StoragePolicyID:            cnsVolumeInfo.Spec.StoragePolicyID,
 					Namespace:                  cnsVolumeInfo.Spec.Namespace,
 					Capacity:                   cnsVolumeInfo.Spec.Capacity,
 					IsStorageQuotaM2FSSEnabled: isStorageQuotaM2FSSEnabled,
 				})
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to create snapshot on volume %q: %v", volumeID, err)
+			}
+			var patch map[string]interface{}
+			if cnsSnapshotInfo.AggregatedSnapshotCapacityInMb == -1 {
+				patchAnnotation := common.MergeMaps(cnsVolumeInfo.Annotations, map[string]string{"csi.vsphere.missing-snapshot-aggregated-capacity": "true"})
```

Comment:
Lets add log message here, saying we couldn't retrieve aggregated snapshot capacity for vol and snapshotit